lib/utils.ex

```diff
@@ -0,0 +1,10 @@
+defmodule ProtoValidator.Utils do
+  @moduledoc ""
+
+  def pipe_validates(objects, validate_fun) do
+    objects
+    |> Stream.map(fn object -> validate_fun.(object) end)
+    |> Stream.filter(&Kernel.match?({:error, _msg}, &1))
```

Comment:
We need to think about whether and when we need to use Stream. If we need to support options like `early_return`, we can use Stream to return early when we have errors and improve the performance. Otherwise, Stream will add some overhead, which is not necessary.